src/v/storage/compaction_reducers.cc

```diff
@@ -120,7 +120,10 @@ copy_data_segment_reducer::filter(model::record_batch&& batch) {
     // offset translation
     if (
       batch.header().type == model::record_batch_type::raft_configuration
-      || batch.header().type == model::record_batch_type::archival_metadata) {
+      || batch.header().type == model::record_batch_type::archival_metadata
+      || batch.header().type == model::record_batch_type::group_abort_tx
+      || batch.header().type == model::record_batch_type::group_commit_tx
+      || batch.header().type == model::record_batch_type::group_prepare_tx) {
```

Comment:
Is this commit `storage: Do not comact group_* log records` unrelated to this PR? There is other work going on related to transactions and compaction. Should this commit be related to a GitHub issue or some other initiative related to that work?

Reply:
It's related. While doing chaos testing for KIP-447 Vadim found a problem, we haven't noticed it before because KIP-447 changed how the clients are allowed to interact with RP. Once this PR is merged Vadim will send a follow up changing the key format of the txn batches fixing the issue and removing this temporary workaround.

The format change isn't related to the KIP-447 so it's gonna be a separate PR.

Reply:
👍